casatables/src/glue.cc

```diff
@@ -131,8 +133,130 @@ extern "C" {
         return -1;
     }
 
+    // TableDesc
+
+    GlueTableDesc *
+    tabledesc_create(
+        // TODO: figure out exactly what this is.
+        // Apparently this is actually the name of the table?
+        const StringBridge &type
+    )
+    {
+        // TODO: expose this?
+        GlueTableDesc::TDOption tdOption = GlueTableDesc::TDOption::New;
+        return new GlueTableDesc(bridge_string(type), tdOption);
+    }
+
+    GlueTableDesc *
+    tabledesc_add_scalar_column(
+        GlueTableDesc &tableDesc,
+        GlueDataType dataType,
+        const StringBridge &columnName,
+        ExcInfo &exc
+    )
+    {
+        try {
+
+            switch (dataType) {
+
+#define CASE(DTYPE, CPPTYPE) \
+            case casacore::DTYPE: { \
+                tableDesc.addColumn(casacore::ScalarColumnDesc<CPPTYPE>( \
+                    bridge_string(columnName) \
+                )); \
+                break; \
+            }
+
+            CASE(TpBool, casacore::Bool)
+            CASE(TpChar, casacore::Char)
+            CASE(TpUChar, casacore::uChar)
+            CASE(TpShort, casacore::Short)
+            CASE(TpUShort, casacore::uShort)
+            CASE(TpInt, casacore::Int)
+            CASE(TpUInt, casacore::uInt)
+            CASE(TpFloat, float)
+            CASE(TpDouble, double)
+            CASE(TpComplex, casacore::Complex)
+            CASE(TpDComplex, casacore::DComplex)
+
+#undef CASE
+
+            case casacore::TpString:
+                throw std::runtime_error("use table_get_scalar_column_data_string for TpString columns");
+
+            default:
+                throw std::runtime_error("unhandled scalar column data type");
+            }
+        } catch (...) {
+            handle_exception(exc);
+        }
+
+        return &tableDesc;
+    }
+
+    // GlueTableDesc *
+    // tabledesc_add_column_array(
+    //     GlueTableDesc &tableDesc,
+    //     GlueDataType dataType,
+    //     const StringBridge &columnName,
+    //     casacore::Int ndim
+    // )
+    // {
+    //     const casacore::ColumnDesc column = new casacore::ArrayColumnDesc<dataType>(
+    //         bridge_string(columnName),
+    //         ndim
+    //     );
+
+    //     tableDesc.addColumn(column);
+    // }
```

Comment:
I think it is better not to add commented-out code in most cases. If you're planning to implement this function later in this PR, though, great!

Reply:
`tabledesc_add_array_column` is coming soon! once i'm happy with `tabledesc_add_scalar_column`

Reply:
done!